src/lib/pipeline/M1PortAllocator.cc

```diff
@@ -20,10 +20,13 @@ uint16_t M1PortAllocator::allocate(const std::vector<uint16_t>& ports) {
   uint16_t bestWeight = 0xFFFF;
 
   uint16_t bestRSQueueSize = 0xFFFF;
-  bool foundRS = false;
+  // Only used in assertions so produces warning in release mode
+  [[maybe_unused]] bool foundRS = false;
 
-  // Update the the reference for number of free spaces in the reservation
+  // Update the reference for number of free spaces in the reservation
   // stations
+  // TODO Declaration shadows a field of 'simeng::pipeline::M1PortAllocator see
```

Comment:
`rsSizes` could be anything; it is a lambda that gets populated at core construction - there are no restrictions on what this should do (not even with a comment or parameter name). It seems like the parameter (`rsFreeSpaces`) should purely be used as an output. Currently the only use case is `DispatchIssueUnit::getRSSizes` which populates the input vector. It takes a bit of digging to figure out how this works/how it should work so I think all this logic should get a second look (maybe not part of this PR?). I can see the motivation for using a reference parameter as output to prevent slow copying, and the need for the lambda use to more easily get info from the dispatchIssueUnit, but it seems messy/confusing in its current form. It at least needs more comments in the headers to explain how it should be used e.g. think of a new dev trying to implement a new port allocator. `rsSizes_` could also be renamed to better reflect its use e.g. getRsSizes at least

For now I have removed the local declaration and replaced it with a `clear()`of the private member to prevent reallocation.